runtime/lua/vim/treesitter/query.lua

```diff
@@ -407,51 +419,53 @@ local predicate_handlers = {
 -- As we provide lua-match? also expose vim-match?
 predicate_handlers['vim-match?'] = predicate_handlers['match?']
 
+--- See |treesitter-directives| |vim.treesitter.query.add_directive()|.
+---
+--- Directive handler receive the following arguments: (match, pattern, source, directive, metadata)
+---@alias TSDirective fun(match: TSMatch, pattern: integer, source: integer|string, directive: (string|integer)[], metadata: TSMetadata)
+
+--- Table for storing metadata for a match. See |vim.treesitter.query.add_directive()|.
 ---@class TSMetadata
 ---@field range? Range
 ---@field conceal? string
 ---@field [integer] TSMetadata
 ---@field [string] integer|string
 
----@alias TSDirective fun(match: TSMatch, _, _, predicate: (string|integer)[], metadata: TSMetadata)
-
--- Predicate handler receive the following arguments
--- (match, pattern, bufnr, predicate)
-
--- Directives store metadata or perform side effects against a match.
--- Directives should always end with a `!`.
--- Directive handler receive the following arguments
--- (match, pattern, bufnr, predicate, metadata)
 ---@type table<string,TSDirective>
 local directive_handlers = {
-  ['set!'] = function(_, _, _, pred, metadata)
-    if #pred >= 3 and type(pred[2]) == 'number' then
+  -- |treesitter-directive-set!|
+  ['set!'] = function(_, _, _, directive, metadata)
+    if #directive >= 3 and type(directive[2]) == 'number' then
       -- (#set! @capture key value)
-      local capture_id, key, value = pred[2], pred[3], pred[4]
+      local capture_id, key, value = directive[2], directive[3], directive[4]
+      assert(type(capture_id) == 'number')
```

Comment:
Doesn't the if condition already check for directive[2] to be number or not

Reply:
you're right, it's unnecessary.